### ISSUE TITLE
Support for locale in WP_Query

### ISSUE THREAD

**Issue by simonwheatley** [simonwheatley/wp-post-term-locale#2]
Once we've added locale data to `wp_posts` and `wp_terms` in #1, we need to be able to query it, which means adding support to `WP_Query`. 

Off the top of my head:
- We need a new `WP_Query` parameter, possibly `locale`
- The `locale` parameter should default to the locale the site is using, if it's not specified
- Will we have an issue if a site is not using multiple locale, but the data ends up with multiple locale in it… in which case some posts will stop being queryable as they are not associated with the locale the site is currently using?

**Comment by grappler** [comment]
I would return all posts regardless of local if it not specified. This makes it easier to query all posts regardless of locale. The alternative being having to list all locales to get all of the posts. This should also solve the third issue.

**Comment by andreasciamanna** [comment]
I agree.
I think we should not set a default for the `locale` parameter, or set the default to `false|null`.
This way `WP_Query` would return all posts by default.

To help multilingual plugins, we could customize the default behaviour of WP_Query by either adding a filter (which change the default value of the `locale` parameter) or just set a constant, as it has been suggested by @simonwheatley during the chat.
This way, multilingual plugins could tell WP_Query to use the default site language, or to the "current" language, when the parameter is not passed.

**Comment by markhowellsmead** [comment]
I agree. A wide, locale-independent query must still work. Querying locale=de_DE returns the appropriate data or an empty data set. Querying with `false|null` returns all data, independent of locale.
